homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from async_timeout import timeout
+
+# import my_pypi_dependency
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.const import DEFAULT_HOST
+from pyweatherflowudp.errors import AddressInUseError, ListenerError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_entry_flow
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = STEP_MANUAL_DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str}
+)
+
+
+async def _async_has_devices(hass: HomeAssistant) -> bool:
+    """Return if there are devices that can be discovered."""
+
+    event = asyncio.Event()
+    host: str = DEFAULT_HOST
+
+    @callback
+    def device_discovered():
+        """Handle a discovered device."""
+        event.set()
+
+    async with WeatherFlowListener(host) as client:
+        client.on(EVENT_DEVICE_DISCOVERED, lambda _: device_discovered())
+        try:
+            async with timeout(10):
+                await event.wait()
+        except asyncio.TimeoutError:
+            return False
+    LOGGER.debug("Discoverd a device")
+    return True
```

Comment:
Still trying to wrap my head around how to test this Async Event based stuff ... any suggestions or other integrations to look at would be helpful :)